realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -279,8 +396,8 @@ public void setBoolean(String fieldName, boolean value) {
     /**
      * Sets the {@code short} value of the given field.
      *
-     * @param fieldName Field name.
-     * @param value Value to insert.
+     * @param fieldName field name.
```

Comment:
This was done in order to match the actual java types. It also has symmetry with the getters `getShort`/`getInt`/`getLong`. The idea of only one "integer" type is a core specific implementation detail I would try to keep away from the ObjectStore API if at all possible.

Also overloaded methods would still be the same method count in both autocomplete/JavaDoc, so imo. it would only reduce readability:

```
setInt(int)
setInt(short)
setInt(long)
```
vs.
```
setInt(int)
setShort(short)
setLong(long)
```